### ISSUE TITLE
internal/gengapic:  remove omit-snippets option

### ISSUE THREAD

**Issue by quartzmo** [googleapis/gapic-generator-go#1332]
The `omit-snippets` generator option is no longer used and should be removed, including all related conditionals in internal/gengapic.

**Comment by jjkoh95** [comment]
Hi @quartzmo 
can I understand if there is any flag to remove snippets?
Thanks

**Comment by quartzmo** [comment]
@jjkoh95 I'm sorry, I don't understand exactly. What do you want to do?

**Comment by jjkoh95** [comment]
Hi @quartzmo 
I wish to generate clients without snippets, is it possible?
I'm not sure if this project is dedicated only to google apis, but I have some protobuf definition that i need to generate its equivalent http clients.

**Comment by quartzmo** [comment]
Yes, the `omit-snippets` flag is documented in the [readme](https://github.com/googleapis/gapic-generator-go?tab=readme-ov-file#invocation). If set to `true` (the default is `false`), snippets will not be generated. However, this flag may be removed in the future. If you indeed use it in production, please comment on this issue to let us know.

**Comment by jjkoh95** [comment]
I believe this flag might have already been removed in the latest version, setting `ommit-snippets` to true still give me the snippets. 
```
github.com/googleapis/gapic-generator-go v0.39.4
```

It is not too painful for me to remove it everytime i generate it, but it seems to be redundant given the example test file is generated altogether.

**Comment by quartzmo** [comment]
Unfortunately, I think you'll just need to keep removing the snippets. To comply with this issue, the motivation for which is to simplify the codebase, the `omit-snippets` flag will someday be removed anyway.
